HW_7/task 7.2.4.js

```diff
@@ -0,0 +1,19 @@
+// 4. У вас есть массив со скобками, предположим [ ‘(‘, ‘)’, ‘(‘, ‘)’, ‘)’], количество элементов и последовательность может быть разной.
+//   Нужно выяснить, у каждой ли скобки есть соответствующая пара (открывающая и закрывающая).
+
+const array = ['(', ')', '(', ')', ')'];
+
+const checkForPair = (arr) => array.filter(el => el === '(').length === arr.filter(el => el === ')').length;
+
+console.log(checkForPair(array))
+
+//   Усложнение: в массиве могут быть вложены еще массивы на разной глубине. (ПОПРОБУЙТЕ МЕТОД .flat(), изучите как он работает с разными глубинами вложенности)
+//   Вернуть нужно всё также есть ли у каждой скобочки соответствующая пара. 
+//   Пример:
+//   const arr = [[['(']], ')', '(', ')', ')', ['(', ['('], [')']]]
+
+const array2 = [[['(']], ')', '(', ')', ')', ['(', ['('], [')']]];
+
+const checkForPair2 = (arr) => arr.flat(Infinity).filter(el => el === '(').length === arr.flat(Infinity).filter(el => el === ')').length;
```

Comment:
Good! Just one moment: you are performing flat(infinity) 2 times. Better to flat before filtering. Yes, function won't be a one-liner, but..))

Reply:
Thanks, I'll keep in mind for the future that the code will work faster in this case